tests/ui_tools/test_popups.py

```diff
@@ -1134,6 +1159,44 @@ def test_keypress_full_raw_message(
             time_mentions=list(),
         )
 
+    @pytest.mark.parametrize("key", keys_for_command("READ_RECEIPTS"))
+    @pytest.mark.parametrize(
+        "server_feature_level",
+        [137, 135],
+        ids=[
+            "read_receipt_supported_zfl",
+            "read_receipt_unsupported_zfl",
+        ],
+    )
+    def test_keypress_read_receipts(
+        self,
+        server_feature_level: int,
+        message_fixture: Message,
+        key: str,
+        widget_size: Callable[[Widget], urwid_Size],
+    ) -> None:
+        msg_info_view = MsgInfoView(
+            self.controller,
+            message_fixture,
+            title="Message Information",
+            topic_links=OrderedDict(),
+            message_links=OrderedDict(),
+            time_mentions=list(),
+        )
+        self.controller.model.server_feature_level = server_feature_level
+
+        if self.controller.model.server_feature_level >= 137:
+            size = widget_size(msg_info_view)
+            msg_info_view.keypress(size, key)
+            self.controller.show_read_receipts.assert_called_once_with(
+                message=message_fixture,
+                topic_links=OrderedDict(),
+                message_links=OrderedDict(),
+                time_mentions=list(),
+            )
+        else:
+            self.controller.show_read_receipts.assert_not_called()
```

Comment:
Thank you for pointing it out! I checked and the keypress function is called when outside the first branch because the function is always pressable even when it's not displayed. So added a boolean variable `show_read_receipts_label` in views.py similar to `show_edit_history_label` which is set as `server_version_level >= 137` 

So now the keypress is not called even when it's outside the first branch when zfl<137 because the bool isn't true.